### PR TITLE
Basic iOS implementation

### PR DESCRIPTION
**Pull request by coenert**
This pr adds all the components available for Mac to the iOS target. I don't really know if it is good enough, especially the ScrollView part, but most of the components are pretty the same as their mac version. (also added some bad demo's for testing the components)

### CODE REVIEW

**Comment by joshaber** [comment]
This is so awesome, thanks :metal: 

I haven't had a chance to properly sit down and look it over yet, but I should in the next day or so.

**Comment by coenert** [comment]
Thanks! really excited about this library :+1:

I saw the components(also in mac) didn't all react on the alpha and hidden property. I fixed this in iOS for now, but i think it should also be fixed in Mac.

File: Few-iOS/Button.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+//  Button.swift
+//  Few
+//
+//  Created by Coen Wessels on 13/03/15.
+//  Copyright (c) 2015 Josh Abernathy. All rights reserved.
+//
+
+import UIKit
+
+private class FewButton: UIButton {
```
Comment (joshaber) [inline Few-iOS/Button.swift]: Does this exist just so the target-action mechanism works? Could we use `TargetActionTrampoline` instead?

File: Few-iOS/Button.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+public class Button: Element {
+    public var title: NSAttributedString
+    public var enabled: Bool
+    public var isDefault: Bool
+    private var action: () -> ()
+    
+    public init(title: String, enabled: Bool = true, isDefault: Bool = false, action: () -> () = { }) {
+        self.title = NSAttributedString(string: title)
+        self.enabled = enabled
+        self.isDefault = isDefault
```
Comment (joshaber) [inline Few-iOS/Button.swift]: `default` doesn't really have an analog on iOS, does it?

File: Few-iOS/Input.swift [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+//  Input.swift
+//  Few
+//
+//  Created by Coen Wessels on 14-03-15.
+//  Copyright (c) 2015 Josh Abernathy. All rights reserved.
+//
+
+import UIKit
+
+private class FewTextField: UITextField {
```
Comment (joshaber) [inline Few-iOS/Input.swift]: Same question about using `TargetActionTrampoline` as above.

**Comment by joshaber** [comment]
All the files in `xcuserdata` can/should be ignored.

Otherwise this is looking really good! I didn't get a chance to checkout the demo yet though.

>I saw the components(also in mac) didn't all react on the alpha and hidden property. I fixed this in iOS for now, but i think it should also be fixed in Mac.

Ah yeah, good call :+1: Opened #34 so I don't forget.

File: Few-iOS/Button.swift (hunk 1 above)
Comment (coenert) [inline Few-iOS/Button.swift]: Yes it works the same, however the problem with `TargetActionTrampoline` is that the `addTarget` method of UIControl doesn't retain its target. So after you create a new `Button` the `TargetActionTrampoline` gets deallocated.

**Comment by coenert** [comment]
@joshaber I added an .gitignore file, so al useless files are deleted.

File: Few-iOS/Button.swift (hunk 1 above)
Comment (coenert) [inline Few-iOS/Button.swift]: Nevermind, i can fix this :satisfied:. If i set the `trampoline` property of the `Button` to the value of the old `Button` it will remain to work after a rerender.

File: FewCore/Component.swift [hunk 4]
```diff
@@ -234,15 +234,19 @@ public class Component<S>: Element {
...
-		if renderQueued { return }
-
-		renderQueued = true
-
-		let observer = CFRunLoopObserverCreateWithHandler(kCFAllocatorDefault, CFRunLoopActivity.Exit.rawValue, 0, 0) { _, _ in
-			self.renderQueued = false
-			self.render()
-		}
-		CFRunLoopAddObserver(CFRunLoopGetMain(), observer, kCFRunLoopCommonModes)
+#if os(iOS)
```
Comment (joshaber) [inline FewCore/Component.swift]: I'm curious why this change was made?

**Comment by joshaber** [comment]
This is great! Just one question left.

File: FewCore/Component.swift (hunk 4 above)
Comment (coenert) [inline FewCore/Component.swift]: Because the enqueueing didn't work on iOS, but i understand that we want to enqueue this on the end on the main run loop :sweat_smile:. I'll fix this with `NSOperationQueue`.

Comment (joshaber) [inline FewCore/Component.swift]: Kinda different. We want to coalesce all updates within the same runloop. But this is fine for now. I'll take a look at why the enqueueing wasn't working.

**Comment by joshaber** [comment]
Thanks so much! :sparkling_heart:
